pkg/controller/stacks/stack/stack.go

```diff
@@ -492,6 +555,136 @@ func (h *stackHandler) isNamespaced() bool {
 	return apiextensions.ResourceScope(h.ext.Spec.PermissionScope) == apiextensions.NamespaceScoped
 }
 
+// syncSATokenSecret function copies service account token secret from custom resource Kubernetes (a.k.a tenant
+// Kubernetes) to Host Cluster. This secret then mounted to stack controller pods so that they can authenticate.
+func (h *stackHandler) syncSATokenSecret(ctx context.Context, owner metav1.OwnerReference,
+	fromSARef corev1.ObjectReference, toSecretRef corev1.ObjectReference) error {
+	// Get the ServiceAccount
+	fromKube := h.kube
+	toKube := h.hostKube
+
+	sa := corev1.ServiceAccount{}
+	err := fromKube.Get(ctx, client.ObjectKey{
+		Name:      fromSARef.Name,
+		Namespace: fromSARef.Namespace,
+	}, &sa)
+
+	if kerrors.IsNotFound(err) {
+		return errors.Wrap(err, errServiceAccountNotFound)
+	}
+	if err != nil {
+		return errors.Wrap(err, errFailedToGetServiceAccount)
+	}
+	if len(sa.Secrets) < 1 {
+		return errors.New(errServiceAccountTokenSecretNotGeneratedYet)
+	}
+	saSecretRef := sa.Secrets[0]
+	saSecretRef.Namespace = fromSARef.Namespace
+	saSecret := corev1.Secret{}
+
+	err = fromKube.Get(ctx, meta.NamespacedNameOf(&saSecretRef), &saSecret)
+
+	if err != nil {
+		return errors.Wrap(err, errFailedToGetServiceAccountTokenSecret)
+	}
+	saSecretOnHost := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:            toSecretRef.Name,
+			Namespace:       toSecretRef.Namespace,
+			OwnerReferences: []metav1.OwnerReference{owner},
+		},
+		Data: saSecret.Data,
+	}
+
+	err = toKube.Create(ctx, saSecretOnHost)
```

Comment:
any reason we'd want to do an update here if it already exists?

Reply:
Service account token secrets [does not change](https://github.com/kubernetes/kubernetes/issues/20165) once created (as long as corresponding SA is not deleted and recreated). So, I don't see a reason we need to update this. 
If Kubernetes somehow supports SA token secret rotation in the future, we need more changes to support this  because this code path (i.e. stack install) is only called once per stack during create.